csm/core/agent/api.py

```diff
@@ -317,11 +320,47 @@ async def set_secure_headers(request, handler):
         SecureHeaders(csp=True).aiohttp(resp)
         return resp
 
+    # The following workaround is required to ensure the ElasticSearch mappings for
+    # `CsmAuditLogModel` will be in place before any audit log entries are generated.  The query
+    # will force the correct mappings to be created on ElasticSearch according to the corresponding
+    # CSM model if they are not yet in place.  This is required because new audit log entries are
+    # added to ElasticSearch by rsyslog and, in case mappings are not present, rsyslog will create
+    # new ones itself.  The newly created mappings will differ from the ones expected by our
+    # business logic, and we will fail to read data from ElasticSearch.  The operation should be
+    # executed only once for performance purposes.
+
+    # Simple flag to keep track of its initialization amidst the sea of static methods in this
+    # module.
+
+    class AuditLogDbInitialized:
+        status = False
+
+    # This function should be invoked before `Log.audit()`.  The call will be placed inside
+    # `rest_middleware()` because it's the closest async parent call to `Log.audit()`.  We load the
+    # database schema from the YAML file again to avoid having to inject the database provider
+    # through the class constructor.  It is a workaround after all.
+
+    @staticmethod
+    async def _initialize_audit_log_db():
+        if CsmRestApi.AuditLogDbInitialized.status:
+            return
+        db_config = Yaml(const.DATABASE_CONF).load()
```

Comment:
Instead of reloading the database config.. Reuse the already loaded config.
In csm_agent.py, db config is already loaded.
`Conf.load(const.DATABASE_INDEX, f"yaml://{const.DATABASE_CONF}")`

In api.py, you can use it as 
`db_config = Conf.get(const.DATABASE_INDEX, "databases>es_db>config")`

Can you try this out?

Reply:
Hi, @udayan-y21, thank you for your suggestion.  I copied this code snippet from `csm/core/agent/csm_agent.py`.  Although I believe that would work, I would rather keep the snippets equal.

Reply:
@udayan-y21 Hi, please provide your opinion on this.

Reply:
as per new guidelines we should be using confstore concept and should not be using  any functions like YAML as much as possible. We dont want to have a code change if in future we move from yaml to json or any other format with similar structure

Reply:
@kupranay  As I have explaned previously, this code snippet is exactly the one used by `csm/core/agent/csm_agent.py`.  Any further changes to it should, at least in theory, be validated in a separate ticket because they were not even introduced to our codebase yet.

Reply:
@kupranay @udayan-y21 Here is the link to the original implementation: https://github.com/Seagate/cortx-manager/blob/tmain/csm/core/agent/csm_agent.py#L53

Reply:
@tb-s  I agree to pranay's comment. I suggest change needs to be added in both places.. The one you made the change and second in `csm/core/agent/csm_agent.py` . This is how we can avoid further modification in this area..

@kupranay @tb-s  Im aprroving the PR due to sprint timeline constraint. But change needs to be added either now or in future sprint.

Reply:
I agree that this change must be done.  I don't believe, though, that this story is the right place to test and validate it, only because the suggestion has not been validated before.  We are not currently using it.

I can prepare a separate PR with the changes proposed.